From ebe49a40c12acfea6562ee455918b2f3dc02777d Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 23 Jun 2026 13:55:55 -0500
Subject: [PATCH] Omit visionOS from sources

---
 mlx/backend/metal/device.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mlx/backend/metal/device.cpp b/mlx/backend/metal/device.cpp
index d678461..b8a4466 100644
--- a/mlx/backend/metal/device.cpp
+++ b/mlx/backend/metal/device.cpp
@@ -35,9 +35,9 @@ constexpr const char* default_mtllib_path = METAL_PATH;
 
 auto get_metal_version() {
   auto get_metal_version_ = []() {
-    if (__builtin_available(macOS 26, iOS 26, tvOS 26, visionOS 26, *)) {
+    if (__builtin_available(macOS 26, iOS 26, tvOS 26, *)) {
       return MTL::LanguageVersion4_0;
-    } else if (__builtin_available(macOS 15, iOS 18, tvOS 18, visionOS 2, *)) {
+    } else if (__builtin_available(macOS 15, iOS 18, tvOS 18, *)) {
       return MTL::LanguageVersion3_2;
     } else {
       return MTL::LanguageVersion3_1;
@@ -832,7 +832,7 @@ bool is_nax_available() {
   auto _check_nax = []() {
     bool can_use_nax = false;
     if (__builtin_available(
-            macOS 26.2, iOS 26.2, tvOS 26.2, visionOS 26.2, *)) {
+            macOS 26.2, iOS 26.2, tvOS 26.2, *)) {
       can_use_nax = true;
     }
     auto& d = metal::device(mlx::core::Device::gpu);
-- 
2.50.1 (Apple Git-155)

