From ebe49a40c12acfea6562ee455918b2f3dc02777d Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 23 Jun 2026 13:55:55 -0500
Subject: [PATCH] Omit visionOS from sources

---
 mlx/backend/metal/device.cpp |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/mlx/backend/metal/device.cpp
+++ b/mlx/backend/metal/device.cpp
@@ -37,7 +37,7 @@ constexpr const char* default_mtllib_pat
 void set_compile_options(
     MTL::CompileOptions* mtl_options,
     const CompileOptions& compile_options) {
-  if (__builtin_available(macOS 15, iOS 18, tvOS 18, visionOS 2, *)) {
+  if (__builtin_available(macOS 15, iOS 18, tvOS 18, *)) {
     switch (compile_options.math_mode) {
       case MathMode::Safe:
         mtl_options->setMathMode(MTL::MathModeSafe);
@@ -64,12 +64,12 @@ void set_compile_options(
 
 auto get_metal_version() {
   auto get_metal_version_ = []() {
-    if (__builtin_available(macOS 27, iOS 27, tvOS 27, visionOS 27, *)) {
+    if (__builtin_available(macOS 27, iOS 27, tvOS 27, *)) {
       // TODO: Use MTL::LanguageVersion4_1 after metal-cpp_27 is released.
       return static_cast<MTL::LanguageVersion>((4 << 16) + 1);
-    } else if (__builtin_available(macOS 26, iOS 26, tvOS 26, visionOS 26, *)) {
+    } else if (__builtin_available(macOS 26, iOS 26, tvOS 26, *)) {
       return MTL::LanguageVersion4_0;
-    } else if (__builtin_available(macOS 15, iOS 18, tvOS 18, visionOS 2, *)) {
+    } else if (__builtin_available(macOS 15, iOS 18, tvOS 18, *)) {
       return MTL::LanguageVersion3_2;
     } else {
       return MTL::LanguageVersion3_1;
@@ -956,7 +956,7 @@ bool is_nax_available() {
   auto _check_nax = []() {
     bool can_use_nax = false;
     if (__builtin_available(
-            macOS 26.2, iOS 26.2, tvOS 26.2, visionOS 26.2, *)) {
+            macOS 26.2, iOS 26.2, tvOS 26.2, *)) {
       can_use_nax = true;
     }
     auto& d = metal::device(mlx::core::Device::gpu);
